### ISSUE TITLE
Resolution preference dropdown should probably be removed?

### ISSUE THREAD

**Issue by mesilliac** [wesnoth/wesnoth#6888]
As i was tidying up the video resizing system i realized that this "resolution" dropdown in preferences is kind of useless?
![Screenshot from 2022-07-20 18-45-08](https://user-images.githubusercontent.com/243646/179915139-d351f1f5-dd71-46d1-8746-83a7f00cf621.png)

It is disabled in fullscreen mode, as fullscreen is always the borderless-desktop-window type of fullscreen. However when in windowed mode there is no need to choose from a list of resolutions, as the user can just resize the window, and the list isn't meaningful either as it describes supported display resolutions.

Perhaps it should just be removed and the other elements rearranged.

**Comment by max-torch** [comment]
If removed, perhaps add an info statement letting the player know what the resolution is at least.. perhaps keep the resolution like this:
> Resolution: 1366x768

But don't make it look like a changeable setting, merely an info statement.

**Comment by Wedge009** [comment]
I may be odd, but I like setting my window size to 'standard' resolutions. Plus certain UI layouts are only triggered at certain resolutions, as you already know.

...but yeah, I suppose there's no technical reason to set specific resolutions. Though it's pretty standard in games, isn't it?

**Comment by knyghtmare** [comment]
I think keeping this resolution changer is a nice feature. I sometimes adjust the windowed mode size to tinker with things. I dont think it should be discarded.

**Comment by Wedge009** [comment]
What kind of weird resolutions? Maybe that's a bug. I don't recall seeing non-standard options in either Windows or Linux and this is what I see on 1.16.5 release in Linux/X11.

![2](https://user-images.githubusercontent.com/13679322/180103760-bba91cdb-d8df-4063-af01-c765ea21f43e.png)

I haven't got a reliable set-up with Wayland so I can't comment on that. Presumably the current 'checking of what the display supports' is dependent on X11 display.

**Comment by mesilliac** [comment]
Wayland doesn't allow apps to change the screen resolution, which i believe is why it gives only one choice here.

I think the ones i have in X11 are what my monitor is reporting it supports as a signal. It does have many of the standard resolutions, but also some that seem quite weird. The largest one is even bigger than the native resolution of the screen, and breaks things when i try to use it.

![res](https://user-images.githubusercontent.com/243646/180104328-e84a91d5-3ded-48a9-b669-9abe87ab2b14.jpg)

The resolution in X11 also doesn't match the theme resolution when using pixel scaling, and there is nothing between 1920x1080 and 3840x2160, which is where all the window sizes are when using pixel scale of 2, which i do.

I believe on MacOS and Wayland it reports the window size according to the desktop / retina scale factor, so if that doesn't match the pixel scale in-game it will also be incorrect with regards to the size the game sees and uses. On Wayland i only have 1920x1080 as an option, which is 3840x2160 with desktop scale factor of 2.

It might even be better to include scaling in the resolution list. Like have options "1280x720@3x" being 3840x2160 output res and 3x pixel scaling. Then the desktop coordinate system (which can be different from both game resolution and rendering resolution) would be irrelevant, and this could be used clearly to test theme and UI sizing, etc, as they work based on the game resolution (in this case 1280x720)

**Comment by mesilliac** [comment]
Also choosing the desktop res ends up being wrong if you have any kind of OS bar or dock, as it has to resize the window to fit the slightly smaller desktop accounting for the bar/dock. I'm not sure if that can be accounted for in the resolution list somehow... or if maybe it should automatically switch to fullscreen mode in that case.

**Comment by mesilliac** [comment]
2655x1464 was probably the current window size. 800x540 is techncially the new minimum allowed resolution, i guess that's why it's listed. I reduced the absolute minimum a while back so that 1920x1080 can use a pixel scale of 2, if selected by the user.

In addition 1152x864, 1280x1024 and 1680x1050 all seem like weird resolutions to me.

**Comment by Wedge009** [comment]
Ah, that makes sense, regarding 800x540.

1280x1024 is definitely a standard resolution even if 5:4 aspect ratio is no longer common. My first flat panels were that resolution. 1680x1050 is also a standard resolution - I'm stuck with that low resolution with the monitors at my office. 1152x864 is less common, but does follow 4:3 standard.

So I think we can say that the resolutions provided in X11 are okay. Maybe the list should only be disabled if Wayland is detected? If there is to be any change at all.

**Comment by mesilliac** [comment]
I no longer think the list should be disabled. Rather i'm thinking maybe it should have a static list of resolutions, in stead of querying what the monitor supports. This would allow using it to select window size on Wayland as well, and also could represent window sizes we expect the game to work well at.

**Comment by Wedge009** [comment]
...I feel the list given in X11 and Windows is fine already. It sounds like Wayland is the special-case here.

But yes, by all means update the title.

**Comment by mesilliac** [comment]
But why should the list be determined by the user's monitor? It has nothing to do with supported display modes. It's not X11 and Windows that are providing that list, it's the actual display hardware.

**Comment by Wedge009** [comment]
(shrug) It's what already works and seems to be convention in other applications. I know Wesnoth is supposed to support 800x600 minimum (or 800x540 now, as you say) and that's about it.

If anything, what the monitor claims to support makes sense to me, since that's what you're displaying Wesnoth on. But if you think that list should be trimmed down and it's not too restrictive, I suppose it wouldn't bother me that much.

(On the other hand, that list may need to be manually updated in the distant future if/when resolutions greater than UHD become common. I think this was the point of not having a static list.)

**Comment by mesilliac** [comment]
The point of opening this wasn't that i want to remove items from the resolution dropdown. If anything i think the selection of items should be expanded. Rather it was that it seems to have been intended in the past to set actual display resolution in fullscreen mode, which is no longer possible at all.

If it is being used to set window size in non-fullscreen mode, it should be appropriate for this task.

As it stands, i would find it vastly more useful if it allowed me to set the game resolution, rather than the window size.

This is hardly a priority, so i intend to write this up as a good-first-issue type enhancement proposal. To do that i need to figure out exactly how it would be ideally used.

**Comment by soliton-** [comment]
I agree a hard coded list makes sense since we don't actually change display resolution anymore. It could be constrained to only show resolutions up to what can actually fit in the current display resolution though.

**Comment by CelticMinstrel** [comment]
Yeah, that makes sense. Put a hard-coded list somewhere, but when displaying it in prefs, filter for "is smaller than current monitor resolution".

**Comment by Wedge009** [comment]
Also note #6127. Depending on how this is resolved that issue may be resolved too.

**Comment by Wedge009** [comment]
grz0's mention refers to the bug I reported in https://github.com/wesnoth/wesnoth/issues/5518#issuecomment-1183747885

**Comment by Wedge009** [comment]
Noting that I found #7436 precisely because I was still using the drop-down menu. Is it really 'useless'?

(Maybe the fact that it's currently broken will be seen as a point in favour of removing the option altogether...)
